# Intro
This is a simple initial test of the project.

In [1]:
import pandas as pd
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB 
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix

In [7]:
df0 = pd.read_csv('Fake.csv')
df1 = pd.read_csv('True.csv')

In [8]:
print(df0.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23481 entries, 0 to 23480
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   title    23481 non-null  object
 1   text     23481 non-null  object
 2   subject  23481 non-null  object
 3   date     23481 non-null  object
dtypes: object(4)
memory usage: 733.9+ KB
None


# Organize data

In [9]:
df0['label'] = 0
df1['label'] = 1

df = pd.concat([df0.iloc[:5000], df1.iloc[:5000]])

train, test = train_test_split(df, test_size=0.2, random_state=42)


train_text = train.text.tolist()
train_labels = train.label.tolist()

test_text = test.text.tolist()
test_labels = test.label.tolist()

# Training

In [10]:
counter = CountVectorizer()

counter.fit(train_text+test_text)

train_counts = counter.transform(train_text)
test_counts = counter.transform(test_text)

#print(counter.vocabulary_)

classifier = MultinomialNB()
classifier.fit(train_counts,train_labels)

MultinomialNB()

# Test

In [12]:
print("Score = ",classifier.score(test_counts, test_labels))

predict = classifier.predict(test_counts)

print(classifier.classes_)

# n: real news, p: fake news
tn, fp, fn, tp = confusion_matrix(test_labels,predict,labels=[0,1]).ravel()

print(tn, fp, fn, tp)
print('precision=',tp/(tp+fp))

Score =  0.9795
[0 1]
987 25 16 972
precision= 0.9749247743229689


# Conclusion

**Surprisingly, simple naive bayse classifier show 98% of accuracy of fake news classification.**
However, let's think harder about the most important goal of this classification.
The most dangerous outcome of fake news is that people believe **fake news as real** and be agitated to do dangerous action, such as to refuse vaccination, cause violance, veto sustainability, etc.
In other word, it's important to keep **precision of real news as high as possible**.
Here, the **precision is 97.5%**, where **only the half of the data are fake news**.
**In real life, there are tons more fake news, which will drop this precision rate**.
Therefore, in the next project, I'm try to make **a stronger filter**.